source/simulator/assembly.cc

```diff
@@ -1816,6 +2808,31 @@ namespace aspect
                                                     data.local_dof_indices,
                                                     system_matrix,
                                                     system_rhs);
+
+    if (advection_field.is_discontinuous(introspection))
```

Comment:
Please leave a comment somewhere around here that says that we're in the following going to copy the DG contributions element by element, rather than using `ConstraintMatrix::copy_local_to_global`, and that that is ok because there are no constraints on discontinuous fields.